src/components/BooksQuery/BooksQuery.tsx

```diff
@@ -1,13 +1,35 @@
-import { SyntheticEvent, useState } from 'react'
+/* eslint-disable prefer-const */
+import { SyntheticEvent, useEffect, useState } from 'react'
 import './BooksQuery.css'
 import { Link } from 'react-router-dom'
-import { useGetBooks } from '../../hooks/useGetBooks'
+import { useAppSelector, useAppDispatch } from '../../redux/hook'
+import { GoogleBooks, selectCachedData } from '../../redux/BookSlice'
+import { BooksApiResponse } from '../../types/GoogleBooks'
 const BooksQuery = () => {
   const [bookName, setBookName] = useState<string>('')
-  const { books, loading, error, searchBooks } = useGetBooks('react')
+  const { searchInput, FetchedBooks, loading, error } =
+    useAppSelector(selectCachedData)
+  const dispatch = useAppDispatch()
+  const [bookList, setBooks] = useState<BooksApiResponse[]>()
```

Comment:
This state and this hook are unneeded. To fetch books on the first load, you can add a store slice to check if this is first run

```tsx
const {firstLoad} = useAppSelector(selectFirstLoad)

useEffect(() => {
  if (firstLoad) {
    dispatch(searchBooks(searchInput))
    dispatch(unsetFirstLoad())
  }
}, [])
```

Not the best way, but like I said, this using redux for this use case is not the optimal solution